### PR TITLE
Kernel: Support all the CPUID feature flags

### PR DESCRIPTION
**Pull request by linusg**
| Before | After |
|-|-|
| image | image |

...on the same (QEMU) machine.

This was a lot of fun to learn about. Thanks to @alimpfard, @IdanHo, and @tomuta for all the help! :heart:

### CODE REVIEW

File: AK/ArbitrarySizedEnum.h [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        return (*this & mask) == mask;
+    }
+
+    [[nodiscard]] constexpr bool has_any_flag(ArbitrarySizedEnum<T> const& mask) const
+    {
+        return (*this & mask) != 0u;
+    }
+};
+
+#define MAKE_ARBITRARY_SIZED_ENUM(EnumName, T, ...)                                  \
```
Comment (bgianfo) [inline AK/ArbitrarySizedEnum.h:116]: I think many/most AK macros have a `AK_` prefix, we should consider if we want to do the same here?

File: Kernel/Arch/x86/common/CPUID.cpp [hunk 2]
```diff
@@ -295,14 +295,77 @@ StringView cpu_feature_to_string_view(CPUFeature::Type const& feature)
...
+    if (feature == CPUFeature::PCX_L2I)
+        return "pcx_l2i"sv;
     if (feature == CPUFeature::SYSCALL)
         return "syscall"sv;
+    if (feature == CPUFeature::MP)
+        return "mp"sv;
     if (feature == CPUFeature::NX)
         return "nx"sv;
+    if (feature == CPUFeature::MMXEXT)
+        return "mmext"sv;
```
Comment (bgianfo) [inline Kernel/Arch/x86/common/CPUID.cpp:356]: Is this a typo `mmxext`?

**Review by bgianfo — APPROVED** [review]

File: AK/ArbitrarySizedEnum.h (hunk 1 above)
Comment (linusg) [reply AK/ArbitrarySizedEnum.h]: Good point, done!

File: Kernel/Arch/x86/common/CPUID.cpp (hunk 2 above)
Comment (linusg) [reply Kernel/Arch/x86/common/CPUID.cpp]: Great catch! Fixed :+1:

**Pull request merged**